hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -201,9 +200,12 @@ private[hudi] object HoodieSparkSqlWriter {
             throw new HoodieException(s"hoodie table at $basePath does not exist")
           }
 
+          val tableMetaClient = HoodieTableMetaClient.builder.setConf(sparkContext.hadoopConfiguration).setBasePath(path.get).build()
```

Comment:
does this handle the case where currently there are just deletes on active timeline and the schema resolver returns null for schema? we should just ensure code does not break. in that scenario.

Reply:
+1  to handle this in TableSchemaResolver

Reply:
TableSchemaResolver fetches schema from data file footer and not commit metadata. So its all good. I tested for both COW and MOR table locally to ensure schema is fetched by TableSchemaResolver even if last commit is delete operation.